sdk/core/Azure.Core/src/RehydrationToken.Serialization.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+#nullable disable
+
+namespace Azure.Core
+{
+    public partial struct RehydrationToken : IJsonModel<RehydrationToken>
+    {
+        internal RehydrationToken DeserializeRehydrationToken(JsonElement element, ModelReaderWriterOptions options)
+        {
+            if (element.ValueKind == JsonValueKind.Null)
+            {
+                return default;
+            }
+            Optional<Guid> id = default;
```

Comment:
Can we just use Guid? to avoid pulling in Optional internal shared source?

Reply:
Makes sense, updated.